src/main/scala/dpla/ingestion3/mappers/providers/Extractor.scala

```diff
@@ -2,40 +2,88 @@ package dpla.ingestion3.mappers.providers
 
 import java.net.URI
 
-import dpla.ingestion3.model.{DplaMapData, EdmAgent, OreAggregation}
-import dpla.ingestion3.utils.Utils
+import dpla.ingestion3.model.{EdmAgent, OreAggregation}
+import org.apache.commons.codec.digest.DigestUtils
 
-import scala.util.{Failure, Success, Try}
+import scala.util.Try
 
 /**
   * Interface that all provider extractors implement.
   */
 
 trait Extractor {
   // Base item uri
-  private val baseItemUri = "http://dp.la/api/items/"
+  private val baseDplaItemUri = "http://dp.la/api/items/"
 
   def build(): Try[OreAggregation]
   def agent: EdmAgent
+
+  /**
+    * Does the provider use a prefix (typically their provider abbreviation) to
+    * salt the identifier?
+    *
+    * @return Boolean
+    */
+  def useProviderName(): Boolean
+
+  /**
+    * Extract the record's "persistent" identifier. Implementations should raise
+    * an Exception if no ID can be extracted
+    *
+    * @return String Record identifier
+    */
+  def getProviderId(): String
+
   /**
-    * Build the base ID to be hashed. Implemented per provider
+    * The value to salt the identifier with. Not all providers
+    * use this. If not, an empty string should be used
```

Comment:
What's the benefit to using an empty string instead of following a Some/None model?

Reply:
we got into trouble passing in options because they weren't getting decomposed and we ended up hashing Some(”id") rather than "id".  When choosing this method I was trying to be as explicit as possible all the way through

I'll take another look at it though to see if there might be another way...

Reply:
I was thinking that every provider definitely should have a shortname, but `useProviderName()` would determine if it were used in the ID minting.

Reply:
The reason being that the shortname could be useful in other contexts, like automatically building a registry of Extractor implementations, for example.

Reply:
@mdellabitta If that is the case should it read it out of the config file or is that a vector for unreviewed changes that would impact on IDs that we want to actively prevent?

Reply:
Those provider `shortName` are first set by command line args in the HarvesterEntry and then embedded into the harveted DataFrame.  So if we want to rely on those we **could** have an issue where we change the shortName during invocation b/c they changed names w/o realizing that it is going to disrupt the ID minting.  I don't disagree with making every provider have a short name value set for ID minting and relying on the bool for construction.

I'm trying to argue for: 
1) Clustering all the aspects of an ID together 
2) Hard-coding those values as much as possible so any change would trigger a review.

Reply:
I guess I'm trying to get us to have one place to look when we decide what the shortname of a provider should be. Not having this be standardized is biting us in a number of places in the previous ingestion systems.

The problem you're bringing up is actually another iteration of this general problem. I think we should be able to circumvent  by having somewhere in a constant a list of all the Extractors that are valid, and therefore a list of the valid shortnames is easily obtainable. Then if anything was invoked with an invalid shortname, we should be able to instantly throw up on it, in any entry point.

Reply:
Okay I see what you are saying. We can look to this block to do that validation of the shortName before we even hit the Extractor. There is probably a better way to achieve what you are describing but functionally I think it does exactly what we would want ATM. 

https://github.com/dpla/ingestion3/blob/661c0e33154dee45a343bcac6031e6a1b8802e36/src/main/scala/dpla/ingestion3/MappingEntry.scala#L64-L74

Reply:
Right, I'm going for the DRY approach.

You can see how we've already messed this up with PA, which I seem to remember you fixed in one of your many PRs today. Also we should be able to do this validation in the HarvestEntry as well.

Reply:
it is in this PR ;)

Reply:
Ha!